hvsock_test.go

```diff
@@ -123,10 +130,22 @@ func TestHvSockAddresses(t *testing.T) {
 	u := newUtil(t)
 	cl, sv, addr := clientServer(u)
 
-	sra := (sv.RemoteAddr()).(*HvsockAddr)
-	sla := (sv.LocalAddr()).(*HvsockAddr)
-	cra := (cl.RemoteAddr()).(*HvsockAddr)
-	cla := (cl.LocalAddr()).(*HvsockAddr)
+	sra, ok := (sv.RemoteAddr()).(*HvsockAddr)
+	if !ok {
+		t.Fatalf("expected type %T; got %T", new(HvsockAddr), sv.RemoteAddr())
+	}
```

Comment:
You can make this a lot cleaner by defining a function:

```go
func MustBeType[T any](t *testing.T, v any) T {
	v2, ok := v.(T)
	if !ok {
		t.Fatalf("expected type %T; got %T", *new(T), v)
	}
	return v2
}
```

and using it like:

```go
sra := MustBeType[*HvsockAddr](t, sv.RemoteAddr())
```